### ISSUE TITLE
/usr/bin/ld: cannot find -lcuda

### ISSUE THREAD

**Issue by minertom** [pjreddie/darknet#2024]
I have seen this question asked in a forum about a year ago but with no answer provided.

OS=Ubuntu 18.04, Opencv version = Opencv4.2, Cuda = 10.0, libcudnn= 7.6.4.

All programs work separately. Opencv4 imports with "python3 import cv2" and the sample files for cuda and cudnn run. 

The problem is compiling darknet with OPENCV=1 and GPU=1 (otherwise it compiles with GPU=0).

I realize that the darknet Makefile was written for opencv2 not opencv4 and I have made changes WRT that in the make file.

`ifeq ($(OPENCV), 1)
COMMON+= -DOPENCV
CFLAGS+= -DOPENCV
LDFLAGS+= `pkg-config --libs opencv4`
COMMON+= `pkg-config --cflags opencv4`
endif`

The error from make comes after the "pkg-config --libs opencv4" command to the std output

`pkg-config --libs opencv4` -L/usr/local/cuda-10.0/lib64 -lcuda -lcudart -lcublas -lcurand -lstdc++
/usr/bin/ld: cannot find -lcuda`

none of those libraries are in the opencv4.pc file. There are a lot of files referenced by opencv4.pc but not -lcuda -lcudart -lcublas -lcurand -lstdc++. 

The result for "pkg-config --libs cuda-10.0" is:
-L/usr/local/cuda-10.0/lib64 -lcuda

I attempted to add libraries to the opencv4.pc file but to no avail.

Thank You
Tom

**Comment by minertom** [comment]
Fixed this problem with the help from [https://github.com/pjreddie/darknet/issues/1480](url).
It turned out that when I installed the NVIDIA driver the libcuda.so file was not put in a directory where ldconfig --libs opencv4 could access it.

`cd /usr/local/cuda-10.1/lib64

sudo ln -s  /usr/lib/x86_64-linux-gnu/libcuda.so.418.39 libcuda.so`

Some would say that CUDA would automatically install the correct NVIDIA driver but I found that it always failed, even with blacklisting the nouveau drivers and doing the install at runlevel 3 with no x server running. So, I installed the NVIDIA driver separately.

Of course, my driver was 440.36 and my cuda version was 10.0.

Also, since I loaded opencv4 not opencv2 I had to modify the Makefile for darknet with:

LDFLAGS+= `pkg-config --libs opencv4`
COMMON+= `pkg-config --cflags opencv4`

Running now. Don't know why this happened the way it did.

Tom
